### PR TITLE
Data models

### PR DESCRIPTION
**Pull request by RileyManda**
I have completed the task of implementing data models for processing data according to the requirements: 🌱 

# Implementations

Created the following models:

-  User
-  Like
- Post
- Comment
- Created a method called current_user  in application controller.
- Used the models to insert data by creating user,4 posts and a comment to an existing post:::::

<div style="display: flex; justify-content: space-between;">
  Image 1
  Image 2
</div>

## General Requirements

In addition I have ensured the project meets the following general project requirements:

### Linters

There are [no linter errors](https://github.com/microverseinc/linters-config).

### Gitflow

Used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
### Documentation
Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### Ruby requirements

Followed the list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

Please kindly review my implementation.

Thank you 🙏🏻

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+
+  def update_post_counter
+    author.update(posts_count: author.posts.count)
+  end
+
```
Comment (fobadara) [inline app/models/post.rb:9]: [OPTIONAL] It is a good practice to run methods like this with rails callbacks like `:after_save` and make the method private.

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :post
+  belongs_to :user
+
+  def update_comment_counter
+    post.update(comments_counter: post.comments.count)
+  end
+end
```
Comment (fobadara) [inline app/models/comment.rb:8]: [OPTIONAL] It is a good practice to run methods like this with rails callbacks like `:after_save` and make the method private.

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :post, class_name: 'Post'
+  belongs_to :user, class_name: 'User'
+
+  def update_likes_counter
+    post.update(likes_count: post.likes.count)
+  end
+end
```
Comment (fobadara) [inline app/models/like.rb:8]: [OPTIONAL] It is a good practice to run methods like this with rails callbacks like `:after_save` and make the method private.

**Review by fobadara — APPROVED** [review]
Hi @RileyManda, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code
✅ Good first attempt
✅ Good use of separation of concerns


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (RileyManda) [reply app/models/post.rb:9]: Thank you for the review @fobadara 
I will make sure to  implement your suggestion to add callbacks and make the method private before merging.
Thank you 🙏🏻

**Pull request merged**